plugin/contrib/mesos/pkg/executor/executor_test.go

```diff
@@ -195,3 +214,354 @@ func TestSuicide_WithTasks(t *testing.T) {
 		glog.Infoln("jumps verified") // glog so we get a timestamp
 	}
 }
+
+// TestExecutorRegister ensures that the executor thinks it is connected
+// after Register is called.
+func TestExecutorRegister(t *testing.T) {
+	mockDriver := MockExecutorDriver{}
+	updates := make(chan interface{}, 1024)
+	executor := New(Config{
+		Docker:  dockertools.ConnectToDockerOrDie("fake://"),
+		Updates: updates,
+	})
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+
+	initialPodUpdate := kubelet.PodUpdate{
```

Comment:
may want to set `Source` here so we get a better validation test. @joerg84 will be adding in static pod support soon and we probably want to differentiate between the source that he's going to add, and the default apiserver source that we're emulating.